src/server/pfs/server/api_server.go

```diff
@@ -411,6 +423,10 @@ func (a *apiServer) ModifyFile(server pfs.API_ModifyFileServer) (retErr error) {
 	if err != nil {
 		return err
 	}
+	// ensure commit.Repo is always populated
```

Comment:
Is it simply an error if it is nil?

Reply:
good Q. I think for the meantime it makes sense to aid backwards compatibility with old clients, and then ease the transition in the future. I could be convinced otherwise though

Reply:
It might be cleaner to handle this through the getter for now (GetRepo would set it based on the branch if it wasn't set), then move this to an interceptor later when we move away from the getter pattern.